src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -476,35 +476,36 @@ internal Diagnostic WithProgrammaticSuppression(ProgrammaticSuppressionInfo prog
         /// </summary>
         internal bool HasIntersectingLocation(SyntaxTree tree, TextSpan? filterSpanWithinTree = null)
```

Comment:
Should we have a `PerformanceSensitive` attribute on the method?

Reply:
In the original change, adding the attribute was not easy. In the form we finally arrived at, it seems to make sense. I have now document both performance constraints.